### PR TITLE
Fix bug 1148894: Show featured demos on homepage

### PR DESCRIPTION
**Pull request by openjck**

### CODE REVIEW

File: kuma/landing/views.py [hunk 1]
```diff
@@ -12,9 +12,9 @@
 
 def home(request):
     """Home page."""
-    demos = (Submission.objects
-                       .all_sorted('upandcoming')
-                       .exclude(hidden=True))[:12]
+    demos = (Submission.objects.filter(featured=True)
+                               .exclude(hidden=True)
+                               .order_by('-modified').all()[:12])
```
Comment (jezdez) [inline kuma/landing/views.py]: The `all()` at the end is not needed here.

As there is some infrastructure for sorting is already in place, I wonder if you want to add that type of filtered queryset to the `all_sorted` method as another option? Something like `Submission.objects.all_sorted('freshfeatured')` that then filters on featured and orders by modified as you have it here?

**Comment by openjck** [comment]
I think that's a great idea. Will do.

**Comment by openjck** [comment]
Updated! :smile:

File: kuma/demos/models.py [hunk 2]
```diff
@@ -361,6 +361,10 @@ def all_sorted(self, sort=None):
             return queryset.order_by('-likes_total')
         elif sort == 'upandcoming':
             return queryset.order_by('-likes_recent', '-launches_recent')
+        elif sort == 'recentfeatured':
+            return queryset.filter(featured=True) \
```
Comment (jezdez) [inline kuma/demos/models.py]: Please don't use a backslash but parentheses.

**Comment by openjck** [comment]
Updated! :fireworks:

**Comment by jezdez** [comment]
Nicely done! LGTM

**Comment by jezdez** [comment]
The test breaking is our heisenbug with regard to uploaded files.

**Pull request merged**